init.lua

```diff
@@ -820,6 +824,34 @@ function Form:set_as_inventory_for(player, ctx)
     player:set_inventory_formspec(fs)
 end
 
+-- Declared here to be accessable by render_to_formspec_string
+local fs_process_events
+
+-- Returns a tuple of string and function
+function Form:render_to_formspec_string(player, ctx)
+    local formname = ("flow:subform_%x"):format(next_formname)
+    new_next_formname()
+    local fs, form_info = prepare_form(self, player, formname, ctx or {})
```

Comment:
I think `auto_name_id` should be preserved somehow

Reply:
I'm not sure how I might go about this. Considering that the automatic names _do_ need to be unique, this actually does cause a problem if a parent formspec contains two seperate flow formspecs.

Should this be an argument to `Form:render_to_formspec_string`, do you think we should track it akin to how we do `next_formname` or do you have something else in mind?

Reply:
I don't like the idea of using a global counter since `auto_name_id` wraps around at a much lower number, but I'm also not sure about adding per-player counters.

Reply:
Yeah sure! I like that solution. I think that providing `auto_name_id` as an optional third argument would be wise as well, so it can be overridden by the API user.

Reply:
Not sure if that third argument is needed. I can't think of a usecase for it where it would actually be reliable anyway.

Reply:
What do you think about the latest commit?

Reply:
I think it's okay, though table entries should be removed when players leave the game.

Reply:
Done, and pushed. I'll resolve this convo, but feel free to reopen if you've got further concerns.